### DISCUSSION TITLE
Support var declaration where type is inferred after the declaration.

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/csharplang#7860] [Language Ideas] [answered]
I was refactoring some code and now need to explicitly type a declaration after moving some code outside of a lambda, when I looked at it and wondered if this is feasible a language feature, the variable is named `objects`

```cs
                var start = DateTime.Now;
                var filters = CreateFilters();
                var objects;

                Session.DownloadAdvancedDataSet(DatasetIdentifiers.Advanced.LEARNER_USAGE, filters, true, (data) =>
                {
                    var stop = DateTime.Now;
                    var download_seconds = (stop - start).Seconds;

                    objects = ConvertStreamToRecords(data);
                });

                var table = InitializeDataTable(constring, objects);

                using (var connection = CreateAndOpenDBConnection(constring))
                {
                    TruncateTable(connection);
                    InsertTable(connection, table);
                    InsertNewHistory(connection, download_seconds, table);
                }
```
I want to simply declare `objects` as var because it's type can be inferred a few lines down, in the call to `ConvertStreamToRecords`.

Why is var restricted as it is today, are there perhaps cases (like this) where the inference can be reliably made?

**Comment by HaloFour** [comment, marked as answer]
See: #6200 and #520

**Reply by The-Futurist** [reply]
Very nice, thanks!

**Reply by HaloFour** [reply]
I want to say that there was a conversation on Discord not all that long ago where a team member expressed some support for this.
